### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by Graycemuthui**
### Description
In this project I achieved the following:
- Made sure YAGNI, KISS, and DRY rules and Ruby style guide are respected.
- Follow SOLID principles.
- Paid special attention to the "Single-responsiblity Principle".
- Made sure that the role of main.rb is only to start your code and ask the user for the initial choice of options (or to quit).

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -13,126 +13,136 @@ def initialize
...
-    title = gets.chomp
-    print 'Author:'
-    author = gets.chomp
+    title = get_user_input 'Title:'
+    author = get_user_input 'Author:'
     book = Book.new(title, author)
     @books.push(book)
-    puts 'Book created successfully'
+    puts_message 'Book created successfully'
   end
```
Comment (HAMMAS-SALEEM) [inline app.rb]: - [ ] You are not following the SOLID principal which means each function should have only one responsibility but your function have two responsibilities which are following

- Taking the input (It's great that you are not using gets.chomp multiple times in this file)
- Creating a book

**Please make different function for taking inputs**

File: app.rb [hunk 2]
```diff
@@ -13,126 +13,136 @@ def initialize
...
       @people.push(student)
-      puts 'Student does not have parent permission'
+      puts_message 'Student does not have parent permission'
     when 'y'
       student = Student.new(age, nil, name, true)
       @people.push(student)
-      puts 'Student created successfully'
+      puts_message 'Student created successfully'
     end
   end
```
Comment (HAMMAS-SALEEM) [inline app.rb]: - [ ] You are not following the SOLID principal which means each function should have only one responsibility but your function have two responsibilities which are following

- Taking the input (It's great that you are not using gets.chomp multiple times in this file)
- Creating a student

**Please make different function for taking inputs**

File: app.rb [hunk 3]
```diff
@@ -13,126 +13,136 @@ def initialize
...
-    print 'Specialization:'
-    specialization = gets.chomp
+    age = get_user_input 'Age: '
+    name = get_user_input 'Name:'  
+    specialization = get_user_input 'Specialization:'   
     teacher = Teacher.new(age, specialization, name)
     @people.push(teacher)
-    puts 'Teacher created successfully'
+    puts_message 'Teacher created successfully'
   end
```
Comment (HAMMAS-SALEEM) [inline app.rb]: - [ ] You are not following the SOLID principal which means each function should have only one responsibility but your function have two responsibilities which are following

- Taking the input (It's great that you are not using gets.chomp multiple times in this file)
- Creating a teacher

**Please make different function for taking inputs**

File: app.rb [hunk 4]
```diff
@@ -13,126 +13,136 @@ def initialize
...
+      rental_book = get_rental_book
+      rental_person = get_rental_person
+     
+      date = get_user_input('Date(YYYY/MM/DD):').to_s  
       rental_details = Rental.new(date, @books[rental_book], @people[rental_person])
       @rentals.push(rental_details)
-      puts 'Rental created successfully'
+      puts_message 'Rental created successfully'
     end
   end
```
Comment (HAMMAS-SALEEM) [inline app.rb]: - [ ] You are not following the SOLID principal which means each function should have only one responsibility but your function have two responsibilities which are following

- Taking the input (It's great that you are not using gets.chomp multiple times in this file)
- Creating rental

**Please make different function for taking inputs**

**Review by HAMMAS-SALEEM — CHANGES_REQUESTED** [review]
# Required Changes :recycle:
almost-there

## Hey @Graycemuthui, @bobb-Rob  :wave:

## Highlights :1st_place_medal:

:heavy_check_mark: Following GitFlow
:heavy_check_mark: Following requirements
:heavy_check_mark: Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Descriptive Branch Name
✔️ Following DRY, KISS, YAGNI

Good job on implementing some of the requirements, there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Bondok6 — APPROVED** [review]
# STATUS: APPROVED :green_circle: :clap: 

## Hi @Graycemuthui and @bobb-Rob 🙋‍♂️, 
Your project is complete!  There is nothing else to say other than... Great job 👏 
Congratulations! 🎉 

TO HIGHLIGHT:

- Great job working on the requirements by the previous reviewer ✅
- Your project is professional ✅
- Your repo is professional ✅
- Well structured files ✅


Now you can merge this branch and move on :1st_place_medal: 
HAPPY CODING AND GOOD LUCK WITH THE COMMING ONES :crossed_fingers: :four_leaf_clover: 




Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**